MekHQ/src/mekhq/campaign/parts/equipment/MissingEquipmentPart.java

```diff
@@ -178,149 +188,118 @@ public boolean isAcceptableReplacement(Part part, boolean refit) {
         //they are not acceptable substitutes, so we need to check for that as
         //well
         //http://bg.battletech.com/forums/strategic-operations/(answered)-can-a-lance-for-a-35-ton-mech-be-used-on-a-40-ton-mech-and-so-on/
-        Part newPart = getNewPart();
-        newPart.setUnit(unit);
-        if(part instanceof EquipmentPart) {
-            EquipmentPart eqpart = (EquipmentPart)part;
-            EquipmentType et = eqpart.getType();
-            return (type.equals(et) && getTonnage() == part.getTonnage())
-                    && (size == eqpart.getSize())
-                    && part.getStickerPrice().equals(newPart.getStickerPrice());
+        EquipmentPart newPart = getNewPart();
+        newPart.setEquipmentNum(getEquipmentNum());
+        newPart.setUnit(unit); // CAW: find a way to do this without setting a unit
```

Comment:
Do we want to un-set the unit afterwards?

Reply:
It gets garbage collected after this call. Now that I have a good test I'll probably move the variable cost calculator to a shared class and use that here instead of this tomfoolery.